pandas/_libs/parsers.pyx

```diff
@@ -1179,19 +1179,32 @@ cdef class TextReader:
             return result, na_count
 
         elif is_integer_dtype(dtype):
-            try:
-                result, na_count = _try_int64(self.parser, i, start,
-                                              end, na_filter, na_hashset)
-                if user_dtype and na_count is not None:
-                    if na_count > 0:
-                        raise ValueError(f"Integer column has NA values in column {i}")
-            except OverflowError:
-                result = _try_uint64(self.parser, i, start, end,
-                                     na_filter, na_hashset)
+            if user_dtype and dtype == 'uint64':
+                result = _try_uint64(self.parser, i, start,
+                                     end, na_filter, na_hashset)
                 na_count = 0
-
-            if result is not None and dtype != 'int64':
-                result = result.astype(dtype)
+            else:
+                try:
+                    result, na_count = _try_int64(self.parser, i, start,
+                                                  end, na_filter, na_hashset)
+                    if user_dtype and na_count is not None:
```

Comment:
can this go after the except OverflowError in an `else:` branch?

Reply:
yes, thanks